### ISSUE TITLE
Setting the font for the object language in the interface

### ISSUE THREAD

**Issue by rohlik-hu** [inception-project/inception#2715]
We are annotating historical texts with some strange special characters which require the proper font to be readable. They are UTF8-encoded, but modern standard Unicode fonts are hard to read and lack some special chars from the private use area. Annotators are being slowed down and partially uncertain about some readings.

It would be desirable to be able to change the font of object language strings, including tokens and lemma information, to a user-defined one in the interface. The rest of the interface and annotations should remain in the respective modern font chosen in the web browser. Ideally, the administrator would be able to set the object language font for any given text or corpus.

I think this might be a useful feature for many users who work with texts in rarer scripts or understudied languages.

**Comment by reckart** [comment]
Can it be expected that all users have the font installed on their machines? Would it be necessary or even possible to upload the font to the project so that the user's browsers could load it from the INCEpTION server?

**Comment by rohlik-hu** [comment]
Good point, thank you! Actually, the second solution would be preferable. Generally, one cannot be sure that the users all have the right font at hand, but the administrator should.

Thanks a lot for taking this feature requests into consideration!

**Comment by vintagentleman** [comment]
Hi! May I ask how deep is this feature down the project backlog?

I have literally the same issue as above. I’m trying to adopt INCEpTION for annotating texts in Church Slavonic, which make extensive use of the Cyrillic Extended Unicode blocks to encode e.g. combining superscript characters. Without the proper font, some words become totally illegible or liable to be easily misinterpreted by the annotators.

![ksnip_20220528-174645](https://user-images.githubusercontent.com/26284913/170830588-15f8eba6-e34f-4d41-8637-8342603a773f.png)

**Comment by reckart** [comment]
Which editor are you using? The brat based editor or another one?

**Comment by vintagentleman** [comment]
Yes, the screenshot above is from brat. AnnotatorJS has the same issue.
Meanwhile, the DIAM Debugger appears to render that same sentence just fine:

![ksnip_20220528-180312](https://user-images.githubusercontent.com/26284913/170831060-c8216def-3c71-468b-9226-ed0f77893acc.png)

Still, it would be nice to be able to just drop my TTF file to the project and apply the font to all source text.

**Comment by reckart** [comment]
I have just tried this Chrome Plugin here: https://chrome.google.com/webstore/detail/force-custom-fonts/hckjchjpkmbihoocajjpjajkggbccgee?hl=en

<img width="800" alt="Screenshot 2022-05-28 at 17 15 04" src="https://user-images.githubusercontent.com/1410238/170831444-910e653d-2bd9-443b-8b11-9a86039c94a8.png">

<img width="966" alt="Screenshot 2022-05-28 at 17 15 24" src="https://user-images.githubusercontent.com/1410238/170831454-a2c50fa5-7724-45c3-814a-e026a2e41958.png">

Maybe that could help you across until we have a better solution?

**Comment by vintagentleman** [comment]
That does the trick, client-side at least. Thank you!

**Comment by ajroussel** [comment]
I would recommend the Stylus addon for this purpose: https://github.com/openstyles/stylus.  It's free, open source software that respects user privacy, and it's available for both Chrome and Firefox.

One feature that looks particularly interesting is the ability to export defined styles: So an admin can define the required font substitutions, export these to JSON, and all the users would need to do is install the addon and import the exported JSON file.  I've found that the following snippet works well for Inception:
```css
.brat text {
    font-family: "Junicode" !important;
}
```
